eth/handler.go

```diff
@@ -452,6 +468,18 @@ func (pm *ProtocolManager) handleMsg(p *peer) error {
 		// Filter out any explicitly requested headers, deliver the rest to the downloader
 		filter := len(headers) == 1
 		if filter {
+			// Check for any responses not matching our whitelist
+			if whitelist := pm.whitelist; whitelist != nil {
```

Comment:
No reason to create this whole assign-the-validate check if you don't use a pointer. A map can be nil, and you can iterate a nil map (will be a noop).

Reply:
Done.